core/example/scafacos_example.cpp

```diff
@@ -0,0 +1,311 @@
+#include <Cabana_Types.hpp>
+#include <Cabana_AoSoA.hpp>
+#include <Cabana_MemberTypes.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <cstdlib>
+#include <iostream>
+#include <string>
+
+#include "fcs.h"
+
+//---------------------------------------------------------------------------//
+// Define particle data.
+//---------------------------------------------------------------------------//
+// Spatial dimension.
+const int space_dim = 3;
+
+// Dimension of the NaCl crystal
+const int default_crystal_size = 2;
+
+// User field enumeration. These will be used to index into the data set. Must
+// start at 0 and increment contiguously.
+//
+// NOTE: Users don't have to make this enum (or some other set of integral
+// constants) but it is a nice way to provide meaning to the different data
+// types and values assigned to the particles.
+//
+// NOTE: These enums are also ordered in the same way as the data in the
+// template parameters below.
+enum UserParticleFields
+{
+    PositionX = 0,
+    PositionY,
+    PositionZ,
+    Velocity,
+    Charge,
+    Potential,
+    Field,
+    Index
+};
+
+// Designate the types that the particles will hold.
+using ParticleDataTypes =
+    Cabana::MemberTypes<float,                        // (0) x-position type
+                        float,                        // (1) y-position type
+                        float,                        // (2) z-position type
+                        double[space_dim],            // (3) velocity type
+                        double,			              // (4) charge
+		                double,			              // (5) potential
+		                double[space_dim],		      // (6) electric field values
+                        int                           // (7) status type
+                        >;
+
+// Declare the memory space.
+using MemorySpace = Cabana::HostSpace;
+
+// Declare the length of the internal vectors
+const int VectorLength = 8;
+
+// Set the type for the particle AoSoA.
+using ParticleList = Cabana::AoSoA<ParticleDataTypes,MemorySpace,VectorLength>;
+
+//---------------------------------------------------------------------------//
+// Helper functions.
+//---------------------------------------------------------------------------//
+// Function to intitialize the particles.
+void initializeParticles( ParticleList particles, int crystal_size )
+{
+    auto p_x = particles.slice<PositionX>();
+    auto p_y = particles.slice<PositionY>();
+    auto p_z = particles.slice<PositionZ>();
+    auto v = particles.slice<Velocity>();
+    auto q = particles.slice<Charge>();
+    auto pot = particles.slice<Potential>();
+    auto field = particles.slice<Field>();
+    auto indx = particles.slice<Index>();
+
+    for ( auto idx = 0; idx != particles.size(); ++idx )
```

Comment:
`auto` will deduce `int` and you will get warning about comparing signed and unsigned integers.